# F# 异常处理

> 原文：<https://www.javatpoint.com/f-sharp-exception-handling>

异常处理是处理程序异常终止的标准机制。异常是程序执行过程中发生的一种情况。它可能导致程序异常终止，如 dividebyzero 或 nullpointer。

F# 提供了一套内置的工具来处理异常。

## F# 异常类型

F# 提供了两类例外:

1.  。网络类型异常
2.  F# 异常类型

在这里，我们将只讨论 F# 异常。

# F# 不带尝试/带的示例

```

let ExExample a b =
 let mutable c = 0
 c 
```

在上面的代码示例中，函数在除法后返回整数。如果分母大于 0，则效果良好。如果分母为零，会抛出 DevideByZero 异常，异常终止程序。

输出:

```
System.DivideByZeroException: Attempted to divide by zero.
Stopped due to error.

```

为了处理这种情况，F# 提供了**试配**块。它有助于将可疑代码放入 try block。异常处理程序**用**捕捉异常对象。它提供了有关异常的信息。

现在让我们看看如何处理这种情况。详见下一主题。